std/bitmanip.d

```diff
@@ -1199,9 +1245,108 @@ public:
         }
     }
 
+    /**
+    Checks if all the bits in this `BitArray` are set to true.
+    Complexity: $(BIGOH dim).
+    */
+    @property bool all() const @nogc pure nothrow
+    {
+        if (_len == 0)
+            return false;
+        foreach (i; 0 .. fullWords)
+            if (~_ptr[i] != 0)
+                return false;
+        return (_ptr[fullWords] & endMask) == endMask;
+    }
+
+    ///
+    @system unittest
+    {
+        BitArray empty;
+        assert(!empty.all);
+
+        auto a = BitArray([1]);
+        auto b = BitArray([1, 1, 1, 1, 1]);
+        auto c = BitArray([1, 1, 1, 1, 0, 1, 1]);
+        auto d = BitArray([1, 1, 1, 0]);
+        assert(a.all);
+        assert(b.all);
+        assert(!c.all);
+        assert(!d.all);
+
+        bool[70] bits = 1;
+        bits[65] = 0;
+        auto e = BitArray(bits);
+        assert(!e.all);
+    }
+
+    /**
+    Checks if any of the bits in the `BitArray` is set to true.
+    Complexity: $(BIGOH dim).
+    */
+    @property bool any() const @nogc pure nothrow
+    {
+        if (_len == 0)
+            return false;
+        foreach (i; 0 .. fullWords)
+            if (_ptr[i] != 0)
+                return true;
+        return (_ptr[fullWords] & endMask) != 0;
+    }
+
+    ///
+    @system unittest
+    {
+        BitArray empty;
+        assert(!empty.any);
+
+        auto a = BitArray([0]);
+        auto b = BitArray([0, 0, 0, 1, 0, 1]);
+        auto c = BitArray([1, 0, 0, 0, 0, 0]);
+        auto d = BitArray([0, 0, 0, 0, 0, 0, 0, 0]);
+        assert(!a.any);
+        assert(b.any);
+        assert(c.any);
+        assert(!d.any);
+
+        bool[70] bits;
+        bits[65] = 1;
+        auto e = BitArray(bits);
+        assert(e.any);
+    }
+
+    /**
+    Checks if none of the bits in the `BitArray` is set to true.
+    Complexity: $(BIGOH dim).
+    */
+    @property bool none() const @nogc pure nothrow
+    {
+        return !any;
```

Comment:
Such one liners are contrary to Phobos' vision. Users can easily add the `!` themselves.

Reply:
+1